bpf/lib/nodeport.h

```diff
@@ -798,9 +801,9 @@ static __always_inline bool nodeport_uses_dsr4(const struct ipv4_ct_tuple *tuple
 	return nodeport_uses_dsr(tuple->nexthdr);
 }
 
-static __always_inline bool nodeport_nat_ipv4_needed(struct __ctx_buff *ctx,
-						     __be32 addr,
-						     bool *from_endpoint __maybe_unused)
+/* Returns true if the packet must be SNAT-ed */
```

Comment:
nit: Is this comment needed? `bool snat_v4_needed()` should be self-explanatory?

Reply:
I think this comment makes sense - we have a function `snat_v4_process()` which does both - SNAT and rev-SNAT. This comment emphasizes that `snat_v4_needed()` only returns true if SNAT (but not rev-SNAT) is required.